src/util/lifecycle_utils.js

```diff
@@ -148,6 +148,11 @@ function build_lifecycle_filter(params) {
      * @param {Object} object_info
      */
     return function(object_info) {
+        // fail if object is a temp file/part or a folder object or a versioned object
```

Comment:
i think it is a good check. but we should probably filter those files on the ilm layer if possible

Reply:
I added these code changed because of potential bugs in code like I saw here - https://github.com/noobaa/noobaa-core/pull/9069